src/worker.py

```diff
@@ -110,21 +110,39 @@ def get_synced_msgs(self) -> Optional[Dict[str, BasePacket]]:
                 return sync_msgs
         return None
 
-    def get_most_recent_color_msg(self) -> Optional[BasePacket]:
-        return self._get_most_recent_msg("color")
+    def _add_msgs_from_queue(
+        self, frame_type: Literal["color", "depth"], queue_handler: QueuePacketHandler
+    ) -> None:
+        queue_obj = queue_handler.get_queue()
+        with queue_obj.mutex:
+            q_snapshot = list(queue_obj.queue)
 
-    def get_most_recent_depth_msg(self) -> Optional[BasePacket]:
-        return self._get_most_recent_msg("depth")
+        for msg in q_snapshot:
+            self.add_msg(msg, frame_type, msg.get_sequence_num())
 
     def _get_most_recent_msg(
-        self, frame_type: Literal["color", "depth"]
+        self, q_handler: QueuePacketHandler, frame_type: Literal["color", "depth"]
     ) -> Optional[BasePacket]:
+        self._add_msgs_from_queue(frame_type, q_handler)
+        while len(self.msgs) < 1:
+            self._add_msgs_from_queue(frame_type, q_handler)
+            time.sleep(1)
         # Traverse in reverse to get the most recent
         for msg_dict in reversed(self.msgs.values()):
             if frame_type in msg_dict:
                 return msg_dict[frame_type]
         raise Exception(f"No message of type '{frame_type}' in frame queue.")
 
+    def get_most_recent_color_msg(
+        self, color_q_handler: QueuePacketHandler
+    ) -> Optional[BasePacket]:
+        return self._get_most_recent_msg(color_q_handler, "color")
```

Comment:
Why are we proxying `get_most_recent_color_msg` to `_get_most_recent_msg`?

Reply:
v good point. I originally aliased it thinking there would be difference in logic but it didn't